api/deploy.go

```diff
@@ -400,3 +399,77 @@ func deployRebuild(w http.ResponseWriter, r *http.Request, t auth.Token) error {
 	}
 	return nil
 }
+
+// title: rollback update
+// path: /apps/{appname}/deploy/rollback/update
+// method: PUT
+// consume: application/x-www-form-urlencoded
+// responses:
+//   200: Rollback updated
+//   400: Invalid data
+//   403: Forbidden
+//   404: Not found
+func deployRollbackUpdate(w http.ResponseWriter, r *http.Request, t auth.Token) error {
+	appName := r.URL.Query().Get(":appname")
+	instance, err := app.GetByName(appName)
+	if err != nil {
+		return &tsuruErrors.HTTP{Code: http.StatusNotFound, Message: fmt.Sprintf("App %s not found.", appName)}
+	}
+	canUpdateRollback := permission.Check(t, permission.PermAppUpdateDeployRollback, contextsForApp(instance)...)
+	if !canUpdateRollback {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusForbidden,
+			Message: "User does not have permission to do this action in this app",
+		}
+	}
+	opts := app.DeployOptions{
+		App:  instance,
+		User: t.GetUserName(),
+		Kind: app.DeployRollback,
+	}
+	var imageID string
+	evt, err := event.New(&event.Opts{
+		Target:        appTarget(appName),
+		Kind:          permission.PermAppUpdate,
+		Owner:         t,
+		CustomData:    opts,
+		Allowed:       event.Allowed(permission.PermAppReadEvents, contextsForApp(instance)...),
+		AllowedCancel: event.Allowed(permission.PermAppUpdateEvents, contextsForApp(instance)...),
+		Cancelable:    true,
+	})
+	if err != nil {
+		return err
+	}
+	defer func() { evt.DoneCustomData(err, map[string]string{"image": imageID}) }()
+	img := r.FormValue("image")
+	if img == "" {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusBadRequest,
+			Message: "you must specify an image",
+		}
+	}
+	rb := r.FormValue("enabled")
+	rollback, err := strconv.ParseBool(rb)
+	if err != nil {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusBadRequest,
+			Message: fmt.Sprintf("Status `enabled` set to: %s instead of `true` or `false`", rb),
+		}
+	}
+	reason := r.FormValue("reason")
+	if (reason == "") && (rollback == false) {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusBadRequest,
+			Message: "Reason cannot be empty while enabling rollback",
+		}
+	}
+
+	err = app.RollbackUpdate(appName, img, reason, rollback)
+	if err != nil {
+		return &tsuruErrors.HTTP{
+			Code:    http.StatusNotFound,
```

Comment:
I dont think all errors from `app.RollbackUpdate` should be treated as `http.StatusNotFound`